axelor-production/src/main/resources/views/ManufOrder.xml

```diff
@@ -429,12 +452,21 @@
 	<action-attrs name="action-manuf-order-attrs-price-digits">
 		<attribute name="scale" for="costPrice" expr="eval: __config__.app.getNbDecimalDigitForUnitPrice()"/>
 	</action-attrs>
+	
+	<action-attrs name="action-attrs-manuf-order-other-reason">
+		<attribute name="value" for="cancelReasonStr" expr="eval: null"/>
+		<attribute for="cancelReasonStr" name="hidden" expr="cancelReason.freeText == false"/>
+	</action-attrs>
 
 	<action-validate name="action-manuf-order-validate-planned-end-date"
 	   model="com.axelor.apps.production.db.ManufOrder">
 	   <alert message="The planned end date is inferior to the actual date"
 	       if="plannedEndDateT.isBefore(__config__.app.getTodayDateTime().toLocalDateTime())"/>
 	</action-validate>
+	
+	<action-condition name="action-manuf-order-cancel-reason-check">
```

Comment:
No need to add this action, as its same as the one for sale order, we can use it directly.